secret/scanner.go

```diff
@@ -72,6 +141,34 @@ func (s Scanner) Scan(args ScanArgs) types.Secret {
 	}
 }
 
+func removeAllowedSecrets(allRuleResults []RuleResult, allowedLocations [][]int) []RuleResult {
+	if len(allowedLocations) > 0 {
+		results := make([]RuleResult, 0)
+		for _, result := range results {
+			found := false
+			if result.EndPosition < allowedLocations[0][0] {
+				results = append(results, result)
+				continue
+			}
+			for _, allowedLocation := range allowedLocations {
+				if allowedLocation[0] > result.EndPosition {
+					break
+				}
+				if result.StartPosition >= allowedLocation[0] && result.EndPosition <= allowedLocation[1] {
+					found = true
+					break
+				}
+			}
+			if !found {
+				results = append(results, result)
+			}
+		}
+		return results
+	} else {
+		return allRuleResults
+	}
```

Comment:
if-statement is large. The early return may be better.

```suggestion
	if len(allowedLocations) == 0 {
		return allRuleResults
	}
	
	results := make([]RuleResult, 0)
	for _, result := range results {
		found := false
		if result.EndPosition < allowedLocations[0][0] {
			results = append(results, result)
			continue
		}
		for _, allowedLocation := range allowedLocations {
		if allowedLocation[0] > result.EndPosition {
				break
			}
			if result.StartPosition >= allowedLocation[0] && result.EndPosition <= allowedLocation[1] {
				found = true
				break
			}
		}
		if !found {
			results = append(results, result)
		}
	}
	return results
```